### DISCUSSION TITLE
ArcWelder PostProcess

### DISCUSSION THREAD

**Discussion by mberlinger3** [supermerill/SuperSlicer#946] [Show and tell]
Hey guys I made a quick post-process script for arcwelder in superslicer on linux. I run fluidd in a container so I didn't really want to have to install octoprint.  This will replace the original g-code file.

```python
#!/usr/bin/python

import sys
import os
import subprocess

path=<path to unarchived root of arcwelder>
binFile = 'bin/ArcWelder'
logFile = 'log.txt'


logPath = os.path.join(path,logFile)
binPath = os.path.join(path,binFile)

sourceFile=sys.argv[1]

out = subprocess.check_output([binPath, sourceFile])

with open(logPath,'wb') as log:
    log.write(out)
```

**Comment by thebrelon** [comment]
hi!
we are currently looking into arc welder with a small group of 3d printers. we use super slicer and would be interrested to see arc welder being integrated into super slicer, kind of like it is in cura, with the ability to tweak settings like resolution.
do you thing that's doable and would devs would dig in it?

thanks 
new to github, don't know much about the uses here, sorry

**Reply by mberlinger3** [reply]
Hey glad to see someone else using arcwelder. I used a postprocessing script as a stopgap solution (with defaults) because it is easy to implement.

I am not sure how cura integrates it but it is more than doable from a development standpoint. There is no reason the binary cannot be included with super slicer and have the options in the GUI. It would be up to the devels to do this but I think there is an active feature request already. 

Alternatively, my script can also be customized by passing more args into the subprocess call (i.e. for resolution adjustments) if needed.

**Comment by supermerill** [comment]
@lordofhyphens is working to integrate it.

**Reply by salocain** [reply]
i put arcwelder bin in the "bin" folder of superSlicer and i just add "post proccesing script" link to arcwelder.

everithing work well ,but no setting ,ok

**Reply by salocain** [reply]
but integrate the "ARC" function it's to be great 

like kisslicer for exemple

**Comment by DzzD** [comment]
I have tried using ArcWelder as a post processing in SuperSlicer and this is definitely a feature that must be added to SuperSlicer with only some few parameters available it will be a must.

From my tiny experience & tests it give : 
- smaller gcode
- smoother print and surface
- faster print
- resolve problem caused by "Printer=>Minimum extrusion length" too low on high definition model

**Comment by inspectionsbybob** [comment]
Arcwelder is easy to implement, just download the standalone version, and then point at it in the post processing section, and it becomes automatic when you save.  It does take some time so it isn't invoked until you save so you don't waste cycles if you forgot an edit...

My Post Processing script is:
C:\Users\Bob\ArcWelder.exe  -g ;
Works just fine....

**Reply by DzzD** [reply]
Yes I know and that's what I already do, but I was meaning fully integrated in GUI with some settings, but also with the produced gcode in the preview gcode window (and the time computed also)

**Reply by supermerill** [reply]
I think it's already in the gcode preview. The gcode preview should read the gcode after the post-process.
It discretize the arc into small strait segment, like a firmware may do. It's not too small to avoid slowing the preview with too many triangles.

**Reply by DzzD** [reply]
But I only see ArcWelder popup cmd when I save the gcode to my SDCard, so I believe it is not executed when generating gcode but only when saving it no ?

**Reply by supermerill** [reply]
mmm, you're right.
When i added support for G0, maybe that was for loading the generated gcode in the standalone gcode-viewer.

**Reply by kubark42** [reply]
@inspectionsbybob, thanks for the tip! On macOS, it worked perfectly to:
- put `ArcWelder` into `~/Library/Application\ Support/SuperSlicer/`  and then ...
- ... add `~/Library/Application\ Support/SuperSlicer/ArcWelder -g ;` into `Print Settings --> Output options --> Post-processing script`. 

One question: what is the `;` doing on the end, though?

**Comment by DzzD** [comment]
I would like to add a recomendation for the G-code viewer : even if G2/G3 curves are rendered with small segments it should only count for one instruction when sliding horizontaly in G-code viewer (this is what Simplify 3d do and it is IMO usefull for user to understand what happen). (sidenote : in the G-code window viewing the text corresponding to the actual line/curve G-code in a popin would be nice to)

![image](https://user-images.githubusercontent.com/12198339/141672958-e0ec43c9-6cac-4ca4-ba6b-4edf3100017f.png)

**Comment by neophyl** [comment]
The gcode for the current cursor section has been added to PS 2.4 so when that's merged in future that should also appear in the gcode viewer in SuSi.  

Basically if it is in new versions of PS then its likely to make it into Susi too in the future.

**Reply by DzzD** [reply]
Sorry I am new to Super Slicer and dont undestand yet well all the connection between all parts of the project like Slic3r and other, so what is SuSi ?

**Reply by neophyl** [reply]
SuSi is the short form of Super Slicer.  People us PS as a shorthand for Prusa Slicer but naturally Merill didnt like SS being used as a shorthand for Super Slicer given the historical use of that short form in many parts of the world.  So SuSi/Susi

**Reply by DzzD** [reply]
Ha ok :) and I suppose PS is for Prusa Slicer.

So SuperSlicer is based on Prusa Slicer wich is based on Slic3r that's it ?

**Reply by supermerill** [reply]
yes.

**Comment by inspectionsbybob** [comment]
the last ";" was either in the ArcWelder documentation or a typo, in either case it is in all of my profiles and it works, so I leave it alone....
